docs/query.md

```diff
@@ -0,0 +1,265 @@
+# Zoekt Query Language Guide
+
+This guide explains the Zoekt query language, used for searching text within Git repositories. Zoekt queries allow combining multiple filters and expressions using logical operators, negations, and grouping. Here's how to craft queries effectively.
+
+---
+
+## Syntax Overview
+
+A query is made up of expressions. An **expression** can be:
```

Comment:
It's probably worth mentioning that all expressions are evaluated as regular expressions (except for certain exceptions like if a expression is wrapped in `""`).